src/gleam/http.gleam

```diff
@@ -473,3 +477,112 @@ pub fn set_path(req: Request(body), path: String) -> Request(body) {
     query: query,
   )
 }
+
+fn check_token(token: BitString) {
+  case token {
+    <<"":utf8>> -> Ok(Nil)
+    <<"\s":utf8, _>> -> Error(Nil)
+    <<"\t":utf8, _>> -> Error(Nil)
+    <<"\r":utf8, _>> -> Error(Nil)
+    <<"\n":utf8, _>> -> Error(Nil)
+    <<"\c":utf8, _>> -> Error(Nil)
+    <<"\f":utf8, _>> -> Error(Nil)
+    <<_, rest:bit_string>> -> check_token(rest)
+  }
+}
+
+fn parse_cookie_list(cookie_string) {
+  assert Ok(re) = regex.from_string("[,;]")
+  regex.split(re, cookie_string)
+  |> list.filter_map(
+    fn(pair) {
+      case string.split_once(string.trim(pair), "=") {
+        Ok(tuple("", _)) -> Error(Nil)
+        Ok(tuple(key, value)) -> {
+          let key = string.trim(key)
+          let value = string.trim(value)
+          try _ = check_token(bit_string.from_string(key))
+          try _ = check_token(bit_string.from_string(value))
+          Ok(tuple(key, value))
+        }
+        Error(Nil) -> Error(Nil)
+      }
+    },
+  )
+}
+
+/// Fetch the cookies sent in a request.
+///
+/// Note badly formed cookie pairs will be ignored.
```

Comment:
It's part of the spec to do this, I'll see if I can find a link for it and add to the docs. 

(Worth noting because normally I shy away as far as possible from silently dropping things)